datalad/distribution/publish.py

```diff
@@ -219,14 +219,29 @@ def __call__(dataset=None, to=None, path=None, recursive=False,
 
             lgr.info("Publishing dataset {0} to sibling {1} "
                      "...".format(ds, dest_resolved))
+
             # we now know where to push to:
-            ds.repo.push(remote=dest_resolved,
-                         refspec=ds.repo.get_active_branch(),
-                         set_upstream=set_upstream)
+            push_result = ds.repo.push(remote=dest_resolved,
+                                       refspec=ds.repo.get_active_branch(),
+                                       set_upstream=set_upstream)
+
+            from git.remote import PushInfo as PI
+            for push_info in push_result:
+                if push_info.flags & PI.ERROR == PI.ERROR:
+                    lgr.error(push_info.summary)
+                else:
+                    lgr.info(push_info.summary)
+
             # push annex branch:
             if isinstance(ds.repo, AnnexRepo):
-                ds.repo.push(remote=dest_resolved,
-                             refspec="+git-annex:git-annex")
+                push_result = ds.repo.push(remote=dest_resolved,
+                                           refspec="+git-annex:git-annex")
+                for push_info in push_result:
+                    if push_info.flags & PI.ERROR == PI.ERROR:
+                        lgr.error(push_info.summary)
+                    else:
+                        lgr.info(push_info.summary)
+
```

Comment:
I see two identical pieces of code -- please don't delay refactoring into a helper function